### ISSUE TITLE
./data/train.txt

### ISSUE THREAD

**Issue by kochmc** [lizhengwei1992/Semantic_Human_Matting#3]
Hi, 
Great work so far! Where do i find the 6k training images and the according ./data/train.txt file?
Thanks, Chris

**Comment by lizhengwei1992** [comment]
@kochmc you can find more details of my dataset [here](https://github.com/lizhengwei1992/Fast_Portrait_Segmentation/tree/master/dataset)

**Comment by usalexsantos** [comment]
I had a look on the site you mention about the Dataset but I can't find anything related to this text file.

**Comment by Rimay** [comment]
hello, I am facing the same problem, is there a good solution about the 'train.txt'？

**Comment by xiankgx** [comment]
You can generate the train.txt files yourself. It is just a text file where each line in the file is the full path to a input training RGB image.

**Comment by Jason-xin** [comment]
> You can generate the train.txt files yourself. It is just a text file where each line in the file is the full path to a input training RGB image.

@xiankgx  However, there is alpha images available. https://github.com/aisegmentcn/matting_human_datasets
But, I found that it needs trimap when train this code, So, how can I get trimap?

**Comment by xiankgx** [comment]
```
def generate_trimap(alpha):
    fg = np.array(np.equal(alpha, 255).astype(np.float32))
    # fg = cv.erode(fg, kernel, iterations=np.random.randint(1, 3))
    
    unknown = np.array(np.not_equal(alpha, 0).astype(np.float32))
    
    kernel = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (3, 3))
    unknown = cv2.dilate(unknown, kernel, iterations=np.random.randint(1, 20))
    
    trimap = fg * 255 + (unknown - fg) * 128
    return trimap.astype(np.uint8)
```

From the Deep Image Matting (https://github.com/foamliu/Deep-Image-Matting) repo.

**Comment by xiankgx** [comment]
In this repo, the author is trying to generate trimap from binary mask using some KNN method. The author does that because he couldn't got his hands on alpha matte data. When you have alpha matte data, just use the function above.

**Comment by xuehuachunsheng** [comment]
So... In pretraining stage, the train.txt contains many lines, single of which the format is 
im_path + ' ' + trimap_path + '\n'?

How about the train file format at the finetune stage?

**Comment by tomorrow813** [comment]
> In this repo, the author is trying to generate trimap from binary mask using some KNN method. The author does that because he couldn't got his hands on alpha matte data. When you have alpha matte data, just use the function above.

hi
I have a question about how the mask image is obtained

**Comment by xiankgx** [comment]
> > In this repo, the author is trying to generate trimap from binary mask using some KNN method. The author does that because he couldn't got his hands on alpha matte data. When you have alpha matte data, just use the function above.
> 
> hi
> I have a question about how the mask image is obtained

You should obtain your own dataset with image, segmentation mask.

**Comment by xiankgx** [comment]
> So... In pretraining stage, the train.txt contains many lines, single of which the format is
> im_path + ' ' + trimap_path + '\n'?
> 
> How about the train file format at the finetune stage?

In both stages, it is assumed that you have a folder containing 3 folders: **image**, **trimap**, and **alpha**. Images should be put in the image folder, trimaps should be put in the trimap folder, and alphas should be put in the alpha folder. In addition, you need a text files containing the **basenames** of image files to tell the program which set of files are used for training.
